data/StoreSelectionModel.js

```diff
@@ -105,16 +102,11 @@ export class StoreSelectionModel extends HoistModel {
     //------------------------
     cullSelectionReaction() {
         // Remove recs from selection if they are no longer in store e.g. (due to filtering)
+        // Just cleanup and not mutating the ref.  The records @computed provides all observable state
+        const {store} = this;
         return {
-            track: () => this.store.records,
-            run: (records) => {
-                const storeIds = records.map(it => it.id),
-                    selection = this.ids,
-                    newSelection = intersection(storeIds, selection);
-                if (selection.length !== newSelection.length) {
-                    this.select(newSelection);
-                }
-            }
+            track: () => store.records,
+            run: () => remove(this.ids, id => !store.getById(id))
```

Comment:
By default `store.getById()` does not respect filtering - believe that makes this change problematic as it would no longer cull records excluded via filter from the selection.

Along those same lines, should the call to getById within the records getter should pass true for its `respectFilter` arg? 

Also was unsure if the new comment was as intended - "not mutating the ref" meaning not setting `this.ids` to an new array? Because now we _are_ mutating the array itself. I assume this relates to use of `observable.ref` on the property.

Reply:
good catch here as well -- most important in the records getter -- not sure how I missed this.